sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringFormatArgumentNumberMismatch.cs

```diff
@@ -235,25 +240,47 @@ private static IEnumerable<StringFormatItem> ExtractFormatItems(string format)
                 var curlyBraceCount = 0;
                 var previousChar = '\0';
                 StringBuilder currentFormatItemBuilder = null;
+                var isEscapingOpenCurlyBrace = false;
+                var isEscapingCloseCurlyBrace = false;
                 for (int i = 0; i < format.Length; i++)
                 {
                     var currentChar = format[i];
 
                     if (currentChar == '{')
                     {
-                        curlyBraceCount++;
-                        if (currentFormatItemBuilder == null)
+                        if (previousChar == '{' && !isEscapingOpenCurlyBrace)
+                        {
+                            curlyBraceCount--;
+                            currentFormatItemBuilder = null;
+                            isEscapingOpenCurlyBrace = true;
+                        }
+                        else
                         {
-                            currentFormatItemBuilder = new StringBuilder();
+                            curlyBraceCount++;
+                            isEscapingOpenCurlyBrace = false;
+                            if (currentFormatItemBuilder == null)

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 control flow statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS134)